viz/metrics-api/stat_summary.go

```diff
@@ -113,6 +110,8 @@ func (s *grpcServer) StatSummary(ctx context.Context, req *pb.StatSummaryRequest
 				resultChan <- s.nonK8sResourceQuery(ctx, statReq)
 			} else if isTrafficSplitQuery(statReq.GetSelector().GetResource().GetType()) {
 				resultChan <- s.trafficSplitResourceQuery(ctx, statReq)
+			} else if s.isServiceProfileQuery(statReq) {
```

Comment:
Following the logic from my comment, I think that would mean combining this and the previous branch into one:

```
else if s.isServiceQuery(statReq) {
    resultChan <- s.serviceResourceQuery(ctx, statReq)
```